### PR TITLE
1.20.3/4 support

### PR DESCRIPTION
**Pull request by xGab0**
- updated Minecraft to 1.20.3/4
- updated Fabric loader to 0.15.0
- updated Gradle to 8.5
- imports cleanup
- fixed a mixin

### CODE REVIEW

File: src/main/java/com/ishland/fabric/rsls/mixin/MixinSubtitlesHud.java [hunk 1]
```diff
@@ -16,13 +16,13 @@ public abstract class MixinSubtitlesHud {
...
-    @Shadow public abstract void onSoundPlayed(SoundInstance sound, WeightedSoundSet soundSet);
+    @Shadow public abstract void onSoundPlayed(SoundInstance sound, WeightedSoundSet soundSet, float range);
 
     @Inject(method = "onSoundPlayed", at = @At("HEAD"), cancellable = true)
-    private void onSoundPlayedHandler(SoundInstance sound, WeightedSoundSet soundSet, CallbackInfo ci) {
+    private void onSoundPlayedHandler(SoundInstance sound, WeightedSoundSet soundSet, float range, CallbackInfo ci) {
         if (!this.client.isOnThread()) {
             ci.cancel();
-            this.client.execute(() -> this.onSoundPlayed(sound, soundSet));
+            this.client.execute(() -> this.onSoundPlayed(sound, soundSet, range));
```
Comment (ishland) [inline src/main/java/com/ishland/fabric/rsls/mixin/MixinSubtitlesHud.java:25]: This breaks backward-compatibility for versions earlier than 1.20.3.

File: build.gradle [hunk 2]
```diff
@@ -28,7 +28,7 @@ dependencies {
 	mappings "net.fabricmc:yarn:${project.yarn_mappings}:v2"
 	modImplementation "net.fabricmc:fabric-loader:${project.loader_version}"
 
-    include(implementation(annotationProcessor("io.github.llamalad7:mixinextras-fabric:0.2.0")))
+    //include(implementation(annotationProcessor("io.github.llamalad7:mixinextras-fabric:0.2.0")))
```
Comment (ishland) [inline build.gradle:31]: Remove this line entirely.

File: gradle.properties [hunk 3]
```diff
@@ -3,14 +3,14 @@ org.gradle.jvmargs=-Xmx2G
...
+yarn_mappings=1.20.4+build.1
+loader_version=0.15.0
 
 #Fabric api
-fabric_version=0.90.0+1.20.2
+fabric_version=0.91.1+1.20.4
 
 # Mod Properties
-mod_version = 1.1.1+dev
+mod_version = 1.1.2+dev
```
Comment (ishland) [inline gradle.properties:14]: It is unnecessary to bump the version as 1.1.1 doesn't exist yet.

**Review by ishland — CHANGES_REQUESTED** [review]
Overall looks good aside from some small issues.

File: build.gradle (hunk 2 above)
Comment (xGab0) [reply build.gradle]: resolved in [07191ee](https://github.com/RelativityMC/raise-sound-limit-simplified/pull/7/commits/07191eed6c24466de1475e2674cd9080a78e0670)

File: gradle.properties (hunk 3 above)
Comment (xGab0) [reply gradle.properties]: resolved in [07191ee](https://github.com/RelativityMC/raise-sound-limit-simplified/pull/7/commits/07191eed6c24466de1475e2674cd9080a78e0670)

**Review by ishland — APPROVED** [review]

**Pull request merged**